src/navigation/RootNavigator.js

```diff
@@ -14,22 +14,41 @@
  */
 
 import React from 'react';
+import { StackActions } from '@react-navigation/native';
 
 export const rootNavigatorRef = React.createRef();
 
 const navigate = (screenName, screenProps) =>
   rootNavigatorRef.current?.navigate(screenName, screenProps);
 
 const replace = (screenName, screenProps) =>
-  rootNavigatorRef.current?.replace(screenName, screenProps);
+  rootNavigatorRef.current?.dispatch(StackActions.replace(screenName, screenProps));
 
 const goBack = () => rootNavigatorRef.current?.goBack();
 
 const canGoBack = () => !!rootNavigatorRef.current?.canGoBack();
 
+const getCurrentRouteName = () => {
+  if (!rootNavigatorRef.current?.getRootState()) return '';
+
+  const routes = rootNavigatorRef.current?.getRootState()?.routes ?? [];
+
+  return routes[routes.length - 1]?.name ?? '';
+};
+
+const prevRouteName = () => {
+  if (!rootNavigatorRef.current?.getRootState()) return '';
+
+  const routes = rootNavigatorRef.current?.getRootState()?.routes ?? [];
+
+  return routes[routes.length - 2]?.name ?? '';
+};
```

Comment:
simple current/prev route names - prevroute name is used in the `Back` action to set up the next current route name.. hoping to get rid of this in the next PR